utopia-api/src/property-controls/property-controls.ts

```diff
@@ -21,71 +21,84 @@ export type BaseControlType =
   | 'vector3'
   | 'vector4'
 
-interface AbstractControlDescription<T extends ControlType> {
+export type CheckboxControlDescription = {
+  control: 'checkbox'
   label?: string
-  control: T
-  defaultValue?: unknown
-  visibleByDefault?: boolean
-}
-
-interface AbstractBaseControlDescription<T extends BaseControlType>
-  extends AbstractControlDescription<T> {}
-
-export interface CheckboxControlDescription extends AbstractBaseControlDescription<'checkbox'> {
   defaultValue?: boolean
+  visibleByDefault?: boolean
   disabledTitle?: string
   enabledTitle?: string
 }
 
-export interface ColorControlDescription extends AbstractBaseControlDescription<'color'> {
+export type ColorControlDescription = {
+  control: 'color'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: string
 }
 
 export type AllowedEnumType = string | boolean | number | undefined | null
-export interface BasicControlOption<T> {
+export type BasicControlOption<T> = {
   value: T
   label: string
 }
 
 export type BasicControlOptions<T> = AllowedEnumType[] | BasicControlOption<T>[]
 
-export interface PopUpListControlDescription extends AbstractBaseControlDescription<'popuplist'> {
+export type PopUpListControlDescription = {
+  control: 'popuplist'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: unknown
   options: BasicControlOptions<unknown>
 }
 
-export interface ImportType {
+export type ImportType = {
   source: string // importSource
   name: string | null
   type: 'star' | 'default' | null
 }
 
-export interface ExpressionControlOption<T> {
+export type ExpressionControlOption<T> = {
   value: T
   expression: string
   label?: string
   requiredImport?: ImportType
 }
 
-export interface ExpressionPopUpListControlDescription
-  extends AbstractBaseControlDescription<'expression-popuplist'> {
+export type ExpressionPopUpListControlDescription = {
+  control: 'expression-popuplist'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: ExpressionControlOption<unknown>
   options: ExpressionControlOption<unknown>[]
 }
 
-export interface EulerControlDescription extends AbstractBaseControlDescription<'euler'> {
+export type EulerControlDescription = {
+  control: 'euler'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: [number, number, number, string]
 }
 
-export interface NoneControlDescription extends AbstractBaseControlDescription<'none'> {
+export type NoneControlDescription = {
+  control: 'none'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: never
 }
 
-export interface Matrix3ControlDescription extends AbstractBaseControlDescription<'matrix3'> {
+export type Matrix3ControlDescription = {
+  control: 'matrix3'
+  label?: string
+  visibleByDefault?: boolean
```

Comment:
we can use `// prettier-ignore` to further improve the matrix controls like so:
```ts
// prettier-ignore
defaultValue?: [
  number, number, number,
  number, number, number,
  number, number, number,
]
```

Reply:
done!